src/mcp2515.cpp

```diff
@@ -650,8 +612,19 @@ MCP2515::ERROR MCP2515::sendMessage(const TXBn txbn, const struct can_frame *fra
 
   modifyRegister(txbuf->CTRL, TXB_TXREQ, TXB_TXREQ);
 
-  uint8_t ctrl = readRegister(txbuf->CTRL);
-  if ((ctrl & (TXB_ABTF | TXB_MLOA | TXB_TXERR)) != 0) {
+  // Wait for transmission to complete or timeout (assumes we are not in one-shot
+  // mode). The MCP2515 will try to transmit a frame until successful. It will
+  // report errors along the way, but those are just informational. The chip will
+  // only stop trying once we abort the request after the timeout.
+  uint8_t ctrl = 0x00;
+  uint8_t timeout = 255;
```

Comment:
In my testing, I loaded the bus using cantool's `cansend` utility. I also used their `canbusload` utility to measure the bus utiltization. I logged the number of poll iterations it would take before the transmit was successful at each rate. Here's the results.

  | approx bus usage | worst # of poll iterations |
  |------------------|--------------------|
  | 0%               | 5                  |
  | 45%              | 13                 |
  | 85%              | 33                 |

I think a timeout of 255 should be good enough for most cases.